### PR TITLE
[usage] Handle Stripe customers with zero subscriptions

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

The original log (as an error) has already confused us during support. This handles the case of no subscriptions gracefully.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/usage/pkg/stripe/stripe.go [hunk 1]
```diff
@@ -128,26 +126,32 @@ func (c *Client) findCustomers(ctx context.Context, query string) ([]*stripe.Cus
 	return customers, nil
 }
 
-func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Customer, credits int64) (*UsageRecord, error) {
+func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Customer, credits int64) error {
```
Comment (easyCZ) [inline components/usage/pkg/stripe/stripe.go:129]: drops the return, we were not using it anyway

File: components/usage/pkg/stripe/stripe.go [hunk 2]
```diff
@@ -128,26 +126,32 @@ func (c *Client) findCustomers(ctx context.Context, query string) ([]*stripe.Cus
...
 
 		// nullify any existing usage, but do not set it to negative value - negative invoice doesn't make sense...
 		credits = 0
 	}
 
 	subscriptions := customer.Subscriptions.Data
-	if len(subscriptions) != 1 {
-		return nil, fmt.Errorf("customer has an unexpected number of subscriptions %v (expected 1, got %d)", subscriptions, len(subscriptions))
+	if len(subscriptions) == 0 {
+		logger.Info("Customer does not have a valid (one) subscription. This happens when a customer has cancelled their subscription but still exist as a Customer in stripe.")
```
Comment (easyCZ) [inline components/usage/pkg/stripe/stripe.go:143]: This is where we gracefully handle 0 subs

File: components/usage/pkg/stripe/stripe.go [hunk 3]
```diff
@@ -128,26 +126,32 @@ func (c *Client) findCustomers(ctx context.Context, query string) ([]*stripe.Cus
...
+		return nil
+	}
+	if len(subscriptions) > 1 {
```
Comment (easyCZ) [inline components/usage/pkg/stripe/stripe.go:146]: But we still shout when there are more than 1

**Review by andrew-farries — APPROVED** [review]
Thanks

**Pull request merged**